src/consensus/read_validator.rs

```diff
@@ -93,20 +96,35 @@ impl ReadValidator {
         }
     }
 
-    pub fn process_decided_value(&mut self, value: DecidedValue, sync: &ReadSyncRef) {
+    pub fn process_decided_value(&mut self, value: DecidedValue) -> u64 {
         let height = Self::get_decided_value_height(&value);
-        if height > self.last_height.increment() {
+        let num_committed_values = if height > self.last_height.increment() {
             if (self.buffered_blocks.len() as u32) < self.max_num_buffered_blocks {
                 self.buffered_blocks.insert(height, value);
+                0
             } else {
-                info!(%height, last_height = %self.last_height, "Dropping decided block because buffered block space is full")
+                warn!(%height, last_height = %self.last_height, "Dropping decided block because buffered block space is full");
+                0
             }
         } else if height == self.last_height.increment() {
-            self.commit_decided_value(&value, height, &sync);
-            self.process_buffered_blocks(&sync);
+            self.commit_decided_value(&value, height);
+            let num_buffered_blocks_processed = self.process_buffered_blocks();
+            num_buffered_blocks_processed + 1
         } else {
-            info!(%height, last_height = %self.last_height, "Dropping decided block because height is too low")
-        }
+            debug!(%height, last_height = %self.last_height, "Dropping decided block because height is too low");
+            0
+        };
+        self.statsd_client.gauge_with_shard(
+            self.shard_id,
+            "read_validator.num_buffered_blocks",
+            self.buffered_blocks.len() as u64,
+        );
+        self.statsd_client.count_with_shard(
+            self.shard_id,
+            "read_validator.num_commited_values",
```

Comment:
This is just block height right? Do we need to duplicate it?

Reply:
We don't strictly need it. It would be useful to know if we're receiving the decided values though since it's possible we drop them somewhere before they hit the engine.